### PR TITLE
Larry flu/boot diag

### PR DESCRIPTION
**Pull request by LarryFluke**
Added the -BootDiagnostics switch.

### CODE REVIEW

File: scripts/TunnelAzure.ps1 [hunk 1]
```diff
@@ -82,7 +82,12 @@ Function New-Network {
 
 Function New-TunnelVM {    
     Write-Header "Creating VM '$($Context.VmName)'..."
-    az vm create --location $Context.Location --resource-group $Context.ResourceGroup --name $Context.VmName --image $Context.Image --size $Context.Size --ssh-key-values "$($Context.SSHKeyPath).pub" --public-ip-address-dns-name $Context.VmName --admin-username $Context.Username --vnet-name $Context.VnetName --subnet $Context.SubnetName --only-show-errors | Out-Null
+    az vm create --location $Context.Location --resource-group $Context.ResourceGroup --name $Context.VmName --image $Context.Image --size $Context.Size --ssh-key-values "$($Context.SSHKeyPath).pub" --public-ip-address-dns-name $Context.VmName --admin-username $Context.Username --vnet-name $Context.VnetName --subnet $Context.SubnetName --only-show-errors @AdditionalParams | Out-Null
```
Comment (mikeliddle) [inline scripts/TunnelAzure.ps1]: Did you mean to leave @AdditionalParams in here?

File: scripts/TunnelAzure.ps1 [hunk 2]
```diff
@@ -82,7 +82,12 @@ Function New-Network {
...
+
+    if ($Context.BootDiagnostics) {
+        Write-Header "Enabling boot diagnostics..."
+        az vm boot-diagnostics enable --resource-group $Context.ResourceGroup --name $Context.VmName #--storage $script:Context.AzureStorageAccountName
```
Comment (mikeliddle) [inline scripts/TunnelAzure.ps1]: Is this comment still needed? If not, we should remove it.

File: scripts/TunnelHelpers.ps1 [hunk 3]
```diff
@@ -55,6 +55,8 @@ Class TunnelContext {
     [string] $Subnet = "169.254.0.0/16"
     [string[]] $IncludeRoutes = @()
     [string[]] $ExcludeRoutes = @()
+    [bool] $BootDiagnostics = $false
+    [string] $AzureStorageAccountName
```
Comment (mikeliddle) [inline scripts/TunnelHelpers.ps1]: Is this still being used?

File: scripts/TunnelAzure.ps1 (hunk 1 above)
Comment (LarryFluke) [reply scripts/TunnelAzure.ps1]: Oops.

File: scripts/TunnelAzure.ps1 (hunk 2 above)
Comment (LarryFluke) [reply scripts/TunnelAzure.ps1]: Oops.

**Review by mikeliddle — APPROVED** [review]
lgtm

**Pull request merged**